### PR TITLE
fix(web): copying a code block no longer copies triple backticks

### PR DESCRIPTION
**Pull request by ipanasenko**
In code block, when the very last linebreak is highlighed and copied, it also copies markdown's backticks. This should not happen

<table>
<tr>
 <td>Before
 <td>After
<tr>
 <td>
image

 <td>
image

</table>

=====  TEXT BELOW IS GENERATED BY AI  =======

## Problem

Dragging a selection over the final linebreak of a fenced code block ends the range *after* the closing `pre`. The clipboard serializer then no longer saw the selection as sitting inside the block, so it re-fenced the content as markdown. Copying a shell command that way pasted it wrapped in stray backticks, which breaks the paste into a terminal.

## Fix

`serializeRenderedMarkdownFragment` now scans the fragment first. If the only thing a reader would see in it is one code block (header chrome, `select-none` elements and whitespace do not count), the fragment copies as plain code, matching a selection that never left the `pre`. Prose alongside the block, or a second block, still copies as fenced markdown.

## Surfaces

- **Clients:** web, and desktop by wrapping it. Mobile has no rendered-markdown clipboard path.
- **Providers / contracts:** untouched, this is client-side clipboard serialization only.

## Verification

Added unit coverage in `apps/web/src/markdown-clipboard.test.ts` for the drag-past-the-last-line case and for the fragments that must keep their fences. All 6 cases in that file pass. A workspace `pnpm install` is blocked in my environment by an unrelated registry policy on `@clerk/electron@0.0.37` (pinned on `main`), so I ran the file against the module directly and am leaning on CI for the full suite.

No visual change: the behavior is clipboard content only.

Model: Claude Opus 5. Harness: Claude Code.

> [!NOTE]
> ### Fix copying a rendered code block to not include triple backticks
> - Adds `scanForSoleCodeBlock` to detect when a copied fragment contains only one rendered code block and no other visible content (prose, list markers, images, explicit copy text, etc.).
> - When a fragment is code-only, `serializeRenderedMarkdownFragment` returns the `pre` element's plain text instead of wrapping it in fenced Markdown.
> - Fragments that mix a code block with other content still serialize with fences as before.
> - Adds test fixtures and cases covering list items, checkboxes, file chips, trailing-newline selections, and prose-plus-code combinations.
> - Behavioral Change: `serializeRenderedMarkdownFragment` now trims one trailing newline from the `pre` text in the code-only branch; multi-element fragments are unaffected.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 244415a.</sup>
> 
>

---

> [!NOTE]
> **Low Risk**
> Client-only clipboard serialization in web chat markdown; logic is covered by new unit tests with bounded edge cases around list items and data-markdown-copy.
> 
> **Overview**
> Fixes highlight-and-copy from rendered chat markdown when a selection **extends past a code block’s last line** (range ends after the `pre`), which previously serialized the fragment as fenced markdown and pasted **stray triple backticks** around shell commands.
> 
> **`serializeRenderedMarkdownFragment`** now pre-scans the cloned fragment with **`soleCodeBlock` / `scanForSoleCodeBlock`**: if the only reader-visible content is a single `<pre>`, it returns plain **`textContent`** (trimmed trailing newline), matching a selection that never left the block. Mixed fragments still get fences—e.g. prose beside the block, a second block, bare or checkbox list items (markers count as content), or **`data-markdown-copy`** chips like file links. A code block that is the **entire** list item still copies without fences.
> 
> Tests extend the DOM fakes (`querySelector`, `renderedCodeBlock`) and add cases for these clipboard edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 244415a0732c466b897c62e5be9bee2a27edcf71. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/web/src/markdown-clipboard.ts [hunk 1]
```diff
@@ -256,6 +256,60 @@ function serializeNode(node: Node): string {
...
+function scanForSoleCodeBlock(node: Node, scan: SoleCodeBlockScan): void {
+  for (const child of node.childNodes) {
+    if (scan.other) return;
+    if (child.nodeType === Node.TEXT_NODE) {
+      if ((child.textContent ?? "").trim().length > 0) scan.other = true;
+      continue;
+    }
+    if (child.nodeType !== Node.ELEMENT_NODE) continue;
+    const element = child as Element;
+    if (isSkippedElement(element)) continue;
```
[the remark this answers is not in the document]
Comment (ipanasenko) [reply apps/web/src/markdown-clipboard.ts]: Fixed in 5a4734c0f, though with one rule covering both this and the empty-`li` finding rather than a checkbox-specific check.

Reproduced both: an item with no visible text still emits a marker from `serializeListItem`, so the scan dropped it. Rather than special-casing `input[type="checkbox"]`, the scan now treats an `li` that does **not** contain the code block as content of its own. That covers the checkbox-only item, the empty item, and any other marker-only item.

An `li` that *wraps* the block still does not count, since a selection that never left the `pre` would drop that marker too, which is the behavior this PR is matching.

Regression tests added for both shapes; both fail without the change.

File: apps/web/src/markdown-clipboard.ts [hunk 2]
```diff
@@ -256,6 +256,60 @@ function serializeNode(node: Node): string {
...
+    if (element.tagName === "PRE") {
+      if (scan.pre) scan.other = true;
+      else scan.pre = element;
+      continue;
+    }
+    if (element.tagName === "IMG" || element.tagName === "HR") {
+      scan.other = true;
+      continue;
+    }
+    scanForSoleCodeBlock(element, scan);
```
[the remark this answers is not in the document]
Comment (ipanasenko) [reply apps/web/src/markdown-clipboard.ts:309]: Fixed in 5a4734c0f, with a narrower rule than suggested.

Marking any `ul`/`ol` with a direct `li` as other content would also re-fence a code block that *is* the list item, which is a common shape in agent output and the main case this PR fixes. Instead the scan marks an `li` that does not contain the code block, so the empty-item fragment keeps its fences while a block nested in an item still copies plain.

Covered by `keeps fences when a bare list item sits alongside the code block`.
